### DISCUSSION TITLE
pgsc_calc v2.0.0-alpha.6

### DISCUSSION THREAD

**Discussion by nebfield** [PGScatalog/pgsc_calc#298] [Announcements]
# Changelog

Please note the minimum required nextflow version has been updated to v23.10.0, released in October 2023. Run `nextflow self-update` to upgrade your nextflow version. 

## Improvements

* Migrate our custom python tools to new [`pygscatalog`](https://github.com/PGScatalog/pygscatalog) packages
  * Reference / target intersection now considers allelic frequency and variant missingness to determine PCA eligibility
  * Downloads from PGS Catalog should be faster (async)
  * Packages are now documented 
* Update plink version to alpha 5.10 final #179 
* Add docs describing cloud execution 
* Add correlation test comparing calculated scores against known good scores
* When matching variants, matching logs are now written before scorefiles to improve debugging UX
* Improvements to PCA quality (ensuring low missingness and suitable MAF for PCA-eligble variants in target samples).
    *  This could allow us to implement MAF/missingness filters for scoring file variants in the future. 

## Bug fixes

* Fix ancestry adjustment with VCFs #252
* Fix support for scoring files that only have one effect type column #280 
* Fix adjusting PGS with zero variance (skip them) #283
* Check for reserved characters in sampleset names

<hr /><em>This discussion was created from the release <a href='https://github.com/PGScatalog/pgsc_calc/releases/tag/v2.0.0-alpha.6'>pgsc_calc v2.0.0-alpha.6</a>.</em>

**Comment by smlmbrt** [comment]
Known bug #301, will cause some output names to be incorrect and be fixed in next release.

**Comment by nebfield** [comment]
We're investigating unexpected `pgscatalog.core.lib.pgsexceptions.MatchRateError` in some environments (e.g. UK Biobank on a HPC). This release has been downgraded to a pre-release
